### PR TITLE
Prepare 0.22.1 release

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit prepares the 0.22.1 release, this consists of bumping the version number to 0.22.1. It also adds a prelude release note to provide a brief introduction to the release. When this commit merges it should be tagged as the 0.22.1 release.

### Details and comments

TODO:

- [x] Review and update release notes
- [x] Wait for last bugfix PRs to merge and be backported

### CODE REVIEW

File: releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml
Comment (jakelishman) [inline releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml:6]:
```suggestion
    was removed prematurely from Qiskit Terra 0.22.0 which broke compatibility
```

File: releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml
Comment (jakelishman) [inline releasenotes/notes/fix-pauli-basis-dep-27c0a4506ad38d2c.yaml:13]:
```suggestion
    Fixed the premature removal of the ``pauli_list`` keyword argument of
```

File: releasenotes/notes/fix-vqd-kgt2-1ed95de3e32102c1.yaml
Comment (jakelishman) [inline releasenotes/notes/fix-vqd-kgt2-1ed95de3e32102c1.yaml:5]:
```suggestion
    ``k=2`` eigenvalues were computed. Previously this would fail due to an
```

**Review by jakelishman — COMMENTED** [review]
Strictly we're missing a full stop at the end of every `Fixed ...` link, but I don't know how much we care about fix that.

**Review by jakelishman — APPROVED** [review]
I'm happy whether or not we add the full stops.

**Comment by mtreinish** [comment]
All the open PRs targeting 0.22.1 have merged, so I've removed the `on hold` label and added `automerge`. Once this merges we can tag the 0.22.1 release

**Comment by mtreinish** [comment]
There was a CI issue last night that prevented the jobs used to build and publish the release artifacts from running. I manually ran them this morning so everything should be fine now if you re-run CI or update your branch.

**Pull request merged**
